src/pt/animeq/src/eu/kanade/tachiyomi/animeextension/pt/doramogo/AnimeQ.kt

```diff
@@ -0,0 +1,254 @@
+package eu.kanade.tachiyomi.animeextension.pt.animeq
+
+import android.app.Application
+import android.util.Base64
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.animesource.ConfigurableAnimeSource
+import eu.kanade.tachiyomi.animesource.model.AnimeFilterList
+import eu.kanade.tachiyomi.animesource.model.SAnime
+import eu.kanade.tachiyomi.animesource.model.SEpisode
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.animesource.online.ParsedAnimeHttpSource
+import eu.kanade.tachiyomi.lib.bloggerextractor.BloggerExtractor
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.asJsoup
+import eu.kanade.tachiyomi.util.parallelCatchingFlatMapBlocking
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class AnimeQ : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
+
+    override val name = "AnimeQ"
+
+    override val baseUrl = "https://animeq.blog"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    private val preferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", baseUrl)
+        .add("Origin", baseUrl)
+
+    // ============================== Popular ===============================
+    override fun popularAnimeRequest(page: Int) = GET("$baseUrl/160679", headers)
+
+    override fun popularAnimeSelector() = "div.widget_block:contains(Acessados) a"
+
+    override fun popularAnimeFromElement(element: Element) = SAnime.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        title = element.selectFirst("a img")!!.attr("title")
+        thumbnail_url = element.selectFirst("a img")!!.attr("src")
+    }
+
+    override fun popularAnimeNextPageSelector() = null
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/page/$page", headers)
+
+    override fun latestUpdatesSelector() = "div.ContainerEps > article.EpsItem > a"
+
+    override fun latestUpdatesFromElement(element: Element) = SAnime.create().apply {
+        setUrlWithoutDomain(element.attr("href"))
+        title = element.attr("title")
+        val img = element.selectFirst("div.EpsItemImg > img")
+        thumbnail_url = img?.attr("data-src")?.ifEmpty { img?.attr("src") }
```

Comment:
It would be better to create a helper function like
```kt
private fun Element.imgAttr(): String = when {
    hasAttr("data-src") -> absUrl("data-src")
    else -> absUrl("src")
}
```
and then do `thumbnail_url = img.imgAttr()` instead, since this is done elsewhere in the code

Reply:
@Secozzi thanks for the tip.

I will change to do somethink like this;
```kotlin
    private fun Element.tryGetAttr(vararg attributeKeys: String): String? {
        val attributeKey = attributeKeys.first { hasAttr(it) }
        return attributeKey?.let { attr(attributeKey) }
    }
```